src/components/MapView/Legends/index.tsx

```diff
@@ -117,12 +146,53 @@ function LegendItem({
   opacity: initialOpacity,
   children,
   legendUrl,
+  exposure,
+  extent,
 }: LegendItemProps) {
+  const dispatch = useDispatch();
   const map = useSelector(mapSelector);
+  const analysisExposureLoading = useSelector(
+    isExposureAnalysisLoadingSelector,
+  );
+
+  const { startDate: selectedDate } = useSelector(dateRangeSelector);
+
   const [opacity, setOpacityValue] = useState<number | number[]>(
     initialOpacity || 0,
   );
 
+  const analysisResult = useSelector(analysisResultSelector);
+
+  const runExposureAnalysis = async () => {
+    if (!id) {
+      return;
+    }
+
+    if (!extent) {
+      return;
+    } // hasn't been calculated yet
+
+    if (!selectedDate) {
+      throw new Error('Date must be given to run analysis');
+    }
+
+    const params: AnalysisDispatchParams = {
+      hazardLayer: LayerDefinitions[exposure as LayerKey] as WMSLayerProps,
+      baselineLayer: LayerDefinitions['inform' as LayerKey] as NSOLayerProps, // TODO: Make analysis without baselineLayer.
```

Comment:
Reminder for TODO

Reply:
We left this TODO, since the end plan is to run Exposure Analysis without a baselineLayer. Right now an Analysis depends on a NSOLayer to render table and geojson layer, so we picked a default one which is **inform**